### DISCUSSION TITLE
how to change nicegui to static html like index.html

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#831] [Q&A] [answered]
### Question

i want to make android apk with webview and html from nicegui with flutter.  but is there any way to output html file from nicegui and suppose my nicegui code is like this
```py
from nicegui import ui

with ui.card():
    ui.label('Card content')
    ui.button('Add label', on_click=lambda: ui.label('Click!'))
    ui.timer(1.0, lambda: ui.label('Tick!'), once=True)

ui.run()
```

**Comment by rodja** [comment, marked as answer]
You will need to run the Python code somewhere. We can not extract static html and expect it to run. See https://github.com/zauberzeug/nicegui/discussions/313 for the discussions around that.
